src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CodeActionEndpoint.cs

```diff
@@ -181,19 +181,57 @@ private async Task<IEnumerable<CodeAction>> GetCSharpCodeActionsAsync(RazorCodeA
                 return null;
             }
 
-            var filteredCSharpCodeActions = await FilterCSharpCodeActionsAsync(context, csharpCodeActions, cancellationToken);
+            var csharpCodeActionsWithNames = ExtractCSharpCodeActionNamesFromData(csharpCodeActions);
+            var filteredCSharpCodeActions = await FilterCSharpCodeActionsAsync(context, csharpCodeActionsWithNames, cancellationToken);
             return filteredCSharpCodeActions;
         }
 
-        private async Task<IEnumerable<CodeAction>> FilterCSharpCodeActionsAsync(RazorCodeActionContext context, IEnumerable<CodeAction> csharpCodeActions, CancellationToken cancellationToken)
+        private static Dictionary<string, List<CodeAction>> ExtractCSharpCodeActionNamesFromData(IEnumerable<CodeAction> codeActions)
+        {
+            var codeActionsWithNames = new Dictionary<string, List<CodeAction>>();
```

Comment:
Instead of doing this build a RazorCodeAction that wraps `CodeAction` and provides an identifier member